src/kernel/mpipeline.ml

```diff
@@ -29,7 +29,7 @@ module Cache = struct
     in
     match List.assoc key !cache with
     | state ->
-      cache := List.remove_assoc key !cache;
+      cache := (key, state) :: List.remove_assoc key !cache;
```

Comment:
The only reason I can see for doing this is "it's going to be quicker to access next time".
If that's the case, let's perhaps put a comment saying that before the line.

But also: the list seems be limited to a length of 6; is the change really worth it?

Reply:
It's a quick'n'dirty implementation of a LRU (least recently used) cache: we cache the last 6 recently used instances of the typechecker.

If we didn't move the state to the head of the list, that would implement another caching policy... :man_shrugging: